### PR TITLE
WaitForService added to autoscaling_utils.go

### PR DESCRIPTION
**Pull request by socaa**
Part of #11570 

@piosz

### CODE REVIEW

**Comment by piosz** [comment]
LGTM

File: test/e2e/autoscaling_utils.go [hunk 1]
```diff
@@ -245,6 +247,7 @@ func runServiceAndRCForResourceConsumer(c *client.Client, ns, name string, repli
 			},
 		},
 	})
+	waitForService(c, ns, name, true, startServiceInterval, startServiceTimeout)
```
Comment (piosz) [inline test/e2e/autoscaling_utils.go]: please add expectNoError here

Comment (socaa) [inline test/e2e/autoscaling_utils.go]: done

**Comment by socaa** [comment]
PTAL

@piosz

**Comment by socaa** [comment]
PTAL

@piosz

**Comment by piosz** [comment]
LGTM

**Comment by jszczepkowski** [comment]
CC @jszczepkowski

**Comment by piosz** [comment]
Shippable is flaky. @k8s-oncall could you please merge manually since it's e2e test fix?

**Comment by piosz** [comment]
Thanks @erictune!

**Pull request merged**